### PR TITLE
PortAudio 19.7.0-2 C++ bindings package

### PR DESCRIPTION
**Pull request by volkanorhan**
Adds portaudiocpp package (C++ bindings of portaudio-2.0 library)

### CODE REVIEW

**Comment by Biswa96** [comment]
There is already a [portaudio package][1]. Is not this the same project? If yes, is it possible to add the changes in the existing package.

[1]: https://packages.msys2.org/base/mingw-w64-portaudio

File: mingw-w64-portaudiocpp/PKGBUILD [hunk 1]
```diff
@@ -0,0 +1,68 @@
+# Maintainer: Volkan Orhan <volkanorhn@gmail.com>
+
+_realname=portaudiocpp
+pkgbase=mingw-w64-${_realname}
+pkgname="${MINGW_PACKAGE_PREFIX}-${_realname}"
+epoch=1
```
Comment (MehdiChinoune) [inline mingw-w64-portaudiocpp/PKGBUILD:6]: Why?

File: mingw-w64-portaudiocpp/PKGBUILD [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+pkgver=19.7.0
+_tarver=190700_20210406
+pkgrel=2
```
Comment (MehdiChinoune) [inline mingw-w64-portaudiocpp/PKGBUILD:9]:
```suggestion
pkgrel=1
```

File: mingw-w64-portaudiocpp/PKGBUILD (hunk 1 above)
Comment (volkanorhan) [reply mingw-w64-portaudiocpp/PKGBUILD:6]: Because portaudio package doesn't contain C++ bindings

**Comment by volkanorhan** [comment]
> There is already a [portaudio package](https://packages.msys2.org/base/mingw-w64-portaudio). Is not this the same project? If yes, is it possible to add the changes in the existing package.

portaudio package doesn't contain C++ bindings. This package builds only C++ bindings of the portaudio. Of course this can be build together with portaudio but this is a different library than portaudio. On Linux and Homebrew on macOS, portaudiocpp is a different package.

File: mingw-w64-portaudiocpp/PKGBUILD (hunk 1 above)
Comment (MehdiChinoune) [reply mingw-w64-portaudiocpp/PKGBUILD:6]: I am asking about `epoch=1`, you should read about PKGBUILD syntax.

Comment (volkanorhan) [reply mingw-w64-portaudiocpp/PKGBUILD]: OK, I remowed that line since it is not necessary. Thanks Mehdi.

**Comment by Biswa96** [comment]
> On Linux, portaudiocpp is a different package, too.

Generally ArchLinux package naming scheme is followed here. Their portaudio [PKGBUILD][1] file uses `--enable-cxx` option which adds /usr/lib/libportaudiocpp.so file.

Upstream does not have cmake file in cpp bindings. IMO, the cmake file added here should be contributed to upstream first.

[1]: https://github.com/archlinux/svntogit-packages/blob/packages/portaudio/trunk/PKGBUILD

**Comment by volkanorhan** [comment]
> > On Linux, portaudiocpp is a different package, too.
> 
> Generally ArchLinux package naming scheme is followed here. Their portaudio [PKGBUILD](https://github.com/archlinux/svntogit-packages/blob/packages/portaudio/trunk/PKGBUILD) file uses `--enable-cxx` option which adds /usr/lib/libportaudiocpp.so file.
> 
> Upstream does not have cmake file in cpp bindings. IMO, the cmake file added here should be contributed to upstream first.

There is a similar cmake file to be merged on mainstream, and I contributed some enhancements, too, but it will take too much time to have it on stable release version.
Anyway, I am closing this merge request, and I will make portaudio package to use autotools instead of cmake, because stable version doesn't have cmake file for C++ bindings.

**Comment by Biswa96** [comment]
Portaudio was using cmake since this https://github.com/msys2/MINGW-packages/pull/9173.

**Comment by Biswa96** [comment]
Portaudio was using cmake since this https://github.com/msys2/MINGW-packages/pull/9173. Would you like to wait a little please? Let me test this to existing portaudio package.

**Comment by volkanorhan** [comment]
> wait

OK, I am waiting. Thanks @Biswa96

**Pull request closed without merging**